tests/blackbox.py

```diff
@@ -46,12 +49,106 @@ class AwsTestConfigSetImpl(AwsTestConfig):
     name = 'aws+impl'
 
     # The same as AwsTestConfig but with a WALE_S3_ENDPOINT override.
-    def patch(self, test_name, default_test_bucket):
+    def patch(self, test_name):
         self.monkeypatch.setenv(
             'WALE_S3_ENDPOINT',
             'http+path://s3-us-west-1.amazonaws.com:80')
-        return AwsTestConfig.patch(self, test_name, default_test_bucket)
+        return AwsTestConfig.patch(self, test_name)
+
+
+class AwsInstanceProfileTestConfig(object):
+    name = 'aws+instance-profile'
+
+    def __init__(self, request):
+        self.request = request
+        self.monkeypatch = request.getfuncargvalue('monkeypatch')
```

Comment:
@arschles this code is [monkey-patching](http://stackoverflow.com/questions/5626193/what-is-a-monkey-patch) AWS client libe so the integration tests will get an AWS instance profile. Essentially it's mocking up what AWS would typically return from their server so the tests can try to tests against that.